MIGRATION.md

```diff
@@ -0,0 +1,306 @@
+`msgpack/v5` some functions for the logic tuning were added (see
+[UseLooseInterfaceDecoding](https://pkg.go.dev/github.com/vmihailenco/msgpack/v5#Decoder.UseLooseInterfaceDecoding), [UseCompactInts](https://pkg.go.dev/github.com/vmihailenco/msgpack/v5#Encoder.UseCompactInts) etc), it is still impossible
+to achieve full compliance of behavior between `msgpack/v5` and `msgpack.v2`.
+So we don't go this way. We use standard settings if it possible.
+
+#### Call = Call17
+
+Call requests uses `IPROTO_CALL` instead of `IPROTO_CALL_16`.
+
+So now `Call` = `Call17` and `NewCallRequest` = `NewCall17Request`. A result
+of the requests is an array instead of array of arrays.
+
+#### IPROTO constants
+
+* IPROTO constants have been moved to a separate package [go-iproto](https://github.com/tarantool/go-iproto).
+* `PushCode` constant is removed. To check whether the current response is
+  a push response, use `IsPush()` method of the response iterator instead.
+* `ErrorNo` constant is added to indicate that no error has occurred while
+  getting the response. It should be used instead of the removed `OkCode`.
+  See `ExampleErrorNo`.
+
+#### Request interface
+
+* The method `Code() uint32` replaced by the `Type() iproto.Type`.
+* `Response` method added to the `Request` interface.
+
+#### Request changes
+
+* Requests `Update`, `UpdateAsync`, `UpdateTyped`, `Upsert`, `UpsertAsync` no
+longer accept `ops` argument (operations) as an `interface{}`. `*Operations`
+needs to be passed instead.
+* `Op` struct for update operations made private.
+* Removed `OpSplice` struct.
+* `Operations.Splice` method now accepts 5 arguments instead of 3.
+* `UpdateRequest` and `UpsertRequest` structs no longer accept `interface{}`
+for an `ops` field. `*Operations` needs to be used instead.
+
+#### Response interface
+
+* `Response` is now an interface.
+* Response header stored in a new `Header` struct. It could be accessed via
+  `Header()` method.
+
+#### Response changes
+
+* `ResponseIterator` interface now has `IsPush()` method.
+  It returns true if the current response is a push response.
+* For each request type, a different response type is created. They all
+  implement a `Response` interface. `SelectResponse`, `PrepareResponse`,
+  `ExecuteResponse`, `PushResponse` are a part of a public API.
+  `Pos()`, `MetaData()`, `SQLInfo()` methods created for them to get specific
+  info. Special types of responses are used with special requests.
+
+#### Future type
+
+* Method `Get` now returns response data instead of the actual response.
+* New method `GetResponse` added to get an actual response.
+* `Future` constructors now accept `Request` as their argument.
+* Methods `AppendPush` and `SetResponse` accepts response `Header` and data
+  as their arguments.
+* Method `Err` was removed because it was causing improper error handling. You
+  You need to check an error from `Get`, `GetTyped` or `GetResponse` with
```

Comment:
```suggestion
* Method `Err` was removed because it was causing improper error handling.
  You need to check an error from `Get`, `GetTyped` or `GetResponse` with
```

Reply:
Thank you, updated.